src/utils/logger.ts

```diff
@@ -0,0 +1,57 @@
+import pino, { stdTimeFunctions } from 'pino';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+type ExtraLogData = {
+    url: string | undefined;
+    method: string | undefined;
+    statusCode: number;
+};
+
+// configuration for pino, a logger library
+const logger = pino({
+    // Pino logger does not output JSON within NextJS middleware function #33898
+    // https://github.com/vercel/next.js/discussions/33898
+    browser: {
+        write(o) {
+            console.log(JSON.stringify(o));
+        },
+    },
+    level: process.env.PINO_LOG_LEVEL || 'info',
+    // doc notes that formatting time in-process impacts logging performance
+    timestamp: stdTimeFunctions.isoTime, // built-in ISO 8601-formatted time in UTC
+});
+
+export function logDataFromNextAPIReqAndRes(
+    req: NextApiRequest,
+    res: NextApiResponse
+): ExtraLogData {
+    return {
+        url: req.url,
+        method: req.method,
+        statusCode: res.statusCode,
+    };
+}
```

Comment:
The purpose of this module is to provide a logger, but here there is a log output builder specifically for request objects in Next.js, which seems a bit unrelated. What do you think about making this function log the output itself, rather than building a logging output?

As an example, I'm thinking somewhere along the lines of:
```
type RequestMethod = "GET" | "POST" | "PUT" | "PATCH";

export function logRequestData(
    url: string,
    method: RequestMethod,
    statusCode: number,
    logLevel: pino.Level,
): void {
    const logData = {
        url: url,
        method: method,
        statusCode: statusCode,
    };

    if(logLevel === 'error') logger.error(logData);
    else if(logLevel === 'info') logger.info(logData);
    else if(logLevel === 'warn') logger.warn(logData);
}
```

This way, the purpose of `logger.ts` is still only to provide utils specific to logging functionality, and we don't have to worry about different Next.js request objects. That logic doesn't need to be handled here.

Reply:
I agree with the change. It does separate the logic better and more cleanly. I can update the code based on this!